examples/scalar_data/geometry_data.py

```diff
@@ -0,0 +1,46 @@
+"""
+Associating data with geometries
+--------------------------------
+
+This example shows how to colour geometries based on a data array.  This
+functionality is available since Cartopy 0.23.
+
+"""
+import matplotlib.pyplot as plt
+
+import cartopy.crs as ccrs
+import cartopy.io.shapereader as shpreader
+
+
+def main():
+    # Load Natural Earth's country shapefiles.
+    shpfilename = shpreader.natural_earth(resolution='110m',
+                                          category='cultural',
+                                          name='admin_0_countries')
+    reader = shpreader.Reader(shpfilename)
+    countries = reader.records()
+
+    # Get hold of the geometry and population estimate from each country's record.
+    geometries = []
+    population_estimates = []
+
+    for country in countries:
+        geometries.append(country.geometry)
+        population_estimates.append(country.attributes['POP_EST'])
+
+    # Set up a figure and an axes with the Eckert VI projection.
+    fig = plt.figure()
+    ax = fig.add_subplot(projection=ccrs.EckertVI())
+
+    # Plot the geometries coloured according to population estimate.
+    art = ax.add_geometries(geometries, crs=ccrs.PlateCarree(),
+                            array=population_estimates, cmap='YlGnBu', norm='log')
```

Comment:
Set a lower limit of 1 million to get a little more color variation?
```suggestion
                            array=population_estimates, cmap='YlGnBu', norm='log', vmin=1e6)
```

Reply:
The mappable won't take _vmin_ so we have to pass it to the `LogNorm`.  I think that's OK and I had already been wondering about limiting the range, so 👍